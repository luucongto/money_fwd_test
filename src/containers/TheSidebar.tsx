import React from 'react'
import { useAppSelector, useAppDispatch } from '../redux/hooks'
import { set, selectSidebarShow } from '../redux/sidebar'
import {
  CCreateElement,
  CSidebar,
  CSidebarNav,
  CSidebarNavDivider,
  CSidebarNavTitle,
  CSidebarMinimizer,
  CSidebarNavDropdown,
  CSidebarNavItem
} from '@coreui/react'

// sidebar nav config
import navigation from './_nav'
const TheSidebar = (): JSX.Element => {
  const dispatch = useAppDispatch()
  const show = useAppSelector(state => selectSidebarShow(state))

  return (
    <CSidebar
      show={show}
      onShowChange={(val: boolean | string |undefined) => dispatch(set(val))}
    >
      <CSidebarNav>

        <CCreateElement
          items={navigation}
          components={{
            CSidebarNavDivider,
            CSidebarNavDropdown,
            CSidebarNavItem,
            CSidebarNavTitle
          }}
        />
      </CSidebarNav>
      <CSidebarMinimizer className='c-d-md-down-none' />
    </CSidebar>
  )
}

export default React.memo(TheSidebar)
