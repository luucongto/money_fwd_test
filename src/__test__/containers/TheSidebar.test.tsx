import React from 'react'
import TheSidebar from '../../containers/TheSidebar'
import { render, screen } from '@testing-library/react'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
// Before your test, do next setting
const middlewares = [] // If you are not using any middleware, leave the array empty or import it and put it in
const mockStore = configureStore(middlewares) // Setting of store based on the middlewares.
const initState = {
  sidebar: {
    sidebarShow: 'responsive'
  }
}
// Create the store to pass as prop in the <Provider>
const store = mockStore(initState)
test('renders TheSidebar without crashing', () => {
  const wrapperResult = shallow(
    <Provider store={store}>
      <TheSidebar />
    </Provider>
  )
  expect(wrapperResult).toMatchSnapshot()
})
