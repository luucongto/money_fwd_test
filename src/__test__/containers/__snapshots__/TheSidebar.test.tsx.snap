// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TheHeader without crashing 1`] = `ShallowWrapper {}`;

exports[`renders TheSidebar without crashing 1`] = `ShallowWrapper {}`;
